### PR TITLE
Feat/simplify uniswap v3

### PR DESCRIPTION
**Pull request by vic-en**
This PR removes the pool caching logic in Uniswap v3.

### CODE REVIEW

File: src/routes/uniswap_v3.ts
Comment (mifeng) [inline src/routes/uniswap_v3.ts:121]: Changing the route would require changes to the client. The goal of this refactor should be to remove the pairs caching code while minimizing changes to the client needed.

File: src/routes/uniswap_v3.ts
Comment (mifeng) [inline src/routes/uniswap_v3.ts:233]: Applying both `parseFloat` and `toSignificant` seems a bit complex. Are both needed here?

**Review by mifeng — CHANGES_REQUESTED** [review]
In addition to the comments here, I couldn't get the `/price` endpoint to work. I passed in base: "DAI", quote: "WETH", amount: 100, side: "buy" but there was no response and no error after about 1 minute.

**Comment by mifeng** [comment]
@vic-en will add 1 more commit to this PR, after which we should merge to `development` and begin testing

**Comment by dennisocana** [comment]
This is 1/3 pull requests we want to merge in the staging phase to be included in this upcoming Friday's release. As discussed during gateway standup, the current version of Uniswap v3 we have right now in production is not working + it's very complex to use.

The changes here are to simplify Uniswap v3 strategy even with outstanding issues, but a better version of what we currently have. Also adding QA as approver to sign-off before merging this in.

**Review by RC-13 — APPROVED** [review]
Approving PR in behalf of QA, issue seen during testing has been address by the last commit.

**Review by mifeng — APPROVED** [review]

**Pull request merged**
